### PR TITLE
TA Internship Final PR

### PR DESCRIPTION
**Pull request by mhamilton723**

### CODE REVIEW

File: cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala [hunk 1]
```diff
@@ -0,0 +1,560 @@
...
+
+import com.microsoft.ml.spark.cognitive._
+import com.microsoft.ml.spark.core.test.base.TestBase
+import org.apache.spark.SparkException
+import org.apache.spark.ml.param.DataFrameEquality
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema
+import org.apache.spark.sql.functions.{col, explode}
+
+class DetectedLanguageSuiteV4 extends TestBase with DataFrameEquality with TextKey {
```
Comment (mhamilton723) [inline cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala:14]: Needs to extend TransformerFuzzing

File: build.sbt [hunk 2]
```diff
@@ -237,7 +238,10 @@ lazy val cognitive = (project in file("cognitive"))
   .enablePlugins(SbtPlugin)
   .dependsOn(core % "test->test;compile->compile")
   .settings((settings ++ Seq(
-    libraryDependencies += ("com.microsoft.cognitiveservices.speech" % "client-sdk" % "1.14.0"),
+    libraryDependencies ++= Seq(
+      "com.microsoft.cognitiveservices.speech" % "client-sdk" % "1.14.0",
+      "com.azure" % "azure-ai-textanalytics" % "5.1.0-beta.7"
```
[the remark this answers is not in the document]
Comment (mhamilton723) [reply build.sbt:244]: ping on this

File: cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala (hunk 1 above)
Comment (suhas92) [reply cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala]: updated

File: cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+}
+
+object TextSentimentV4 extends ComplexParamsReadable[TextSentimentV4]
+
+class TextSentimentV4(override val uid: String)
+  extends TextAnalyticsSDKBase[SentimentScoredDocumentV4]() {
+  logClass()
+
+  def this() = this(Identifiable.randomUID("KeyphraseExtractionV4"))
```
[the remark this answers is not in the document]
Comment (suhas92) [reply cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala]: updated

**Comment by stuartleeks** [comment]
I can't see support for the async Text Analysis endpoint in this PR? With this now present in the existing `TextAnalyze` transformer, I think it would be good to keep parity with this when adding new versions of Text Analysis transformers. For a project we're working on, we got a double benefit from this endpoint for performance. Firstly,  the async endpoint allows for more documents per batch than the individual synchronous endpoints. Secondly, we were able to submit multiple tasks per document. The combined effect of these was a significant reduction in the number of requests we were required to submit, which made a significant difference in throughput against the text analysis API throttling!

**Pull request closed without merging**